src/Compiler/TypedTree/UpdatePrettyTyparNames.fs

```diff
@@ -0,0 +1,30 @@
+﻿/// The Typars of a Val in the signature data should also be pretty named.
+/// This will happen for the implementation file contents, but not for the signature data.
+/// In this module some helpers will traverse the ModuleOrNamespaceType and update all the typars of each found Val.
+module internal FSharp.Compiler.UpdatePrettyTyparNames
+
+open Internal.Utilities.Library
+open FSharp.Compiler.Syntax.PrettyNaming
+open FSharp.Compiler.SyntaxTreeOps
+open FSharp.Compiler.TypedTree
+open FSharp.Compiler.TypedTreeOps
+
+let updateVal (v: Val) =
+    if not (List.isEmpty v.Typars) then
+        let nms = PrettyTypes.PrettyTyparNames (fun _ -> true) List.empty v.Typars
+
+        (v.Typars, nms)
+        ||> List.iter2 (fun tp nm ->
+            if tp.typar_id.idText = unassignedTyparName then
+                tp.typar_id <- ident (nm, tp.Range))
```

Comment:
Looks kinda similar to these two. If we can also use `PrettyTypes.NeedsPrettyTyparName` maybe these could be unified.

https://github.com/dotnet/fsharp/blob/1bb1c3c930c0dc015d86172c1bdbe7992a15ddcb/src/Compiler/Checking/PostInferenceChecks.fs#L119-L121

https://github.com/dotnet/fsharp/blob/1bb1c3c930c0dc015d86172c1bdbe7992a15ddcb/src/Compiler/Optimize/Optimizer.fs#L607-L609

Reply:
Done!